stdlib/public/runtime/ErrorObject.mm

```diff
@@ -334,6 +332,10 @@ static id _swift_bridgeErrorToNSError_(SwiftError *errorObject) {
   NSDictionary *userInfo =
     swift_stdlib_getErrorUserInfoNSDictionary(value, type, witness);
 
+  // Never produce an empty userInfo dictionary.
+  if (!userInfo)
+    userInfo = SWIFT_LAZY_CONSTANT(@{});
```

Comment:
Do we already do this somewhere else? We might want to factor this out into a static helper so there's only one global initializer for all empty dictionaries.